rfq/negotiator.go

```diff
@@ -293,6 +293,57 @@ func (n *Negotiator) HandleIncomingBuyRequest(
 	return nil
 }
 
+// HandleOutgoingSellOrder handles an outgoing sell order by constructing sell
+// requests and passing them to the outgoing messages channel. These requests
+// are sent to peers.
+func (n *Negotiator) HandleOutgoingSellOrder(order SellOrder) error {
+	// Query the price oracle for a reasonable ask price. We perform this
+	// query and response handling in a separate goroutine in case it is a
+	// remote service and takes a long time to respond.
+	n.Wg.Add(1)
+	go func() {
+		defer n.Wg.Done()
+
+		// Query the price oracle for an asking price.
+		askPrice, _, err := n.queryAskFromPriceOracle(
+			order.Peer, order.AssetID, order.AssetGroupKey,
+			order.MaxAssetAmount, nil,
+		)
+		if err != nil {
+			err := fmt.Errorf("negotiator failed to handle price "+
+				"oracle response: %w", err)
+			n.cfg.ErrChan <- err
+			return
+		}
+
+		request, err := rfqmsg.NewSellRequest(
+			*order.Peer, order.AssetID, order.AssetGroupKey,
+			order.MaxAssetAmount, askPrice,
+		)
+		if err != nil {
+			err := fmt.Errorf("unable to create sell request "+
+				"message: %w", err)
+			n.cfg.ErrChan <- err
+			return
+		}
+
+		// Send the response message to the outgoing messages channel.
+		var msg rfqmsg.OutgoingMsg = request
+		sendSuccess := fn.SendOrQuit(
+			n.cfg.OutgoingMessages, msg, n.Quit,
+		)
+		if !sendSuccess {
+			err := fmt.Errorf("negotiator failed to add sell " +
+				"request message to the outgoing messages " +
+				"channel")
+			n.cfg.ErrChan <- err
+			return
+		}
+	}()
+
+	return nil
```

Comment:
There's a TODO in this function that will end up with an error return code path:
```
// TODO(ffranr): Ensure that we have a suitable buy offer for the asset
//  that our peer is trying to sell to us.
```

It will end up looking like `HandleIncomingBuyRequest`.